pandas/core/dtypes/cast.py

```diff
@@ -409,6 +410,14 @@ def maybe_promote(dtype, fill_value=np.nan):
     elif is_datetime64tz_dtype(dtype):
         if isna(fill_value):
             fill_value = NaT
+        elif not isinstance(fill_value, datetime):
+            dtype = np.dtype(np.object_)
+        elif fill_value.tzinfo is None:
+            dtype = np.dtype(np.object_)
+        elif not tz_compare(fill_value.tzinfo, dtype.tz):
+            # TODO: sure we want to cast here?
```

Comment:
What is your apprehension with this comment?

Reply:
actually you likley remove line 415 as None vs a tz will also not compare.

Reply:
> What is your apprehension with this comment?

For something like setitem, using a timestamp with a different tz is fine, so wouldn't cast to object.  so im a bit wary about doing this without having any real use cases that get here

Reply:
k